pipelines/services/sync.go

```diff
@@ -0,0 +1,117 @@
+package services
+
+import (
+	"encoding/json"
+	"github.com/jfrog/jfrog-client-go/auth"
+	"github.com/jfrog/jfrog-client-go/http/jfroghttpclient"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/httputils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"net/http"
+	"net/url"
+	"path"
+	"strconv"
+)
+
+type SyncService struct {
+	client *jfroghttpclient.JfrogHttpClient
+	auth.ServiceDetails
+}
+
+func (ss *SyncService) getHttpDetails() httputils.HttpClientDetails {
+	httpDetails := ss.ServiceDetails.CreateHttpClientDetails()
+	return httpDetails
+}
+
+func NewSyncService(client *jfroghttpclient.JfrogHttpClient) *SyncService {
+	return &SyncService{client: client}
+}
+
+// SyncPipelineSource trigger sync for pipeline resource
+func (ss *SyncService) SyncPipelineSource(branch string, repoName string) (int, []byte, error) {
```

Comment:
```suggestion
func (ss *SyncService) SyncPipelineSource(branch string, repoName string) error {
```

Reply:
introduced for handling `pollingExecutor` 
`pollingAction := func() (shouldStop bool, responseBody []byte, err error) `